### ISSUE TITLE
Unexpected token )

### ISSUE THREAD

**Issue by gamut** [eslint/eslint#10369]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 9.11.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// Use this file as a starting point for your project's .eslintrc.
// Copy this file, and add rule overrides as needed.
{
  "extends": [
    "airbnb", "plugin:vue/essential",
  ],
  "env": {
    "jquery": true,
    "browser": true
  },
  "globals": {
    "App": true,
    "$": true
  },
  "rules": {
    "max-len": [2, 140, 2],
    "no-param-reassign": "warn",
    "no-underscore-dangle": "off",
    "prefer-destructuring": [
      "error",
      {
        "array": false,
        "object": false
      }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import Vue from 'vue';

export default {
  getChat(token, cb, cbError) {
    const url = Routes.api_v1_chat_path(token);
    Vue.http.get(url).then(response => cb(response.body), () => cbError());
  },

  getChats(cb, cbError) {
    const url = Routes.api_v1_chats_path();
    Vue.http.get(url).then(response => cb(response.body), () => cbError());
  },

  endChat(token, cb, cbError) {
    const url = Routes.api_v1_chat_path(token);
    Vue.http.delete(url).then(response => cb(response.body), () => cbError());
  },

  chatCable: null,
  chatsCable: null,

  disconnectIfNecessary() {
    if (this.chatCable) this.chatCable.unsubscribe();
    this.chatCable = null;
  },

  initChatCable(store, token) {
    this.disconnectIfNecessary();
    if (!token) {
      return;
    }
    this.disconnectIfNecessary();
    this.chatCable = App.cable.subscriptions.create(
      {
        channel: 'ChatChannel',
        chatToken: token,
      },
      {
        connected() {
          this.subscribe();
        },

        disconnected() {
          console.log('WebSocket disconnected.');
        },

        received(data) {
          const message = JSON.parse(data.message);
          const prompt = JSON.parse(data.prompt);
          const chat = JSON.parse(data.chat);

          store.commit('receiveMessage', {
            token,
            chat,
            message,
            prompt,
          });
        },

        sendMessage(message) {
          this.perform('send_message', { message });
        },

        sendResponse(response) {
          this.perform('send_response', { response });
        },

        subscribe() {
          // return this.perform('subscribe');
        },
      },
    );  // THIS IS LINE 72
  },

  disconnectChatsCable() {
    if (this.chatsCable) this.chatsCable.unsubscribe();
    this.chatsCable = null;
  },

  initChatsCable(store) {
    this.chatsCable = App.cable.subscriptions.create(
      {
        channel: 'ChatsChannel',
      },
      {
        connected() {
          this.subscribe();
        },

        disconnected() {
        },

        received(data) {
          const chat = JSON.parse(data.chat);
          store.commit('updateChat', { chat });
        },

        subscribe() {},
      },
    );
  },
};
```

```bash
eslint app/javascript/packs/rather_chat/api/chats.js 
```

**What did you expect to happen?**

This is a perfectly valid file.

**What actually happened? Please include the actual, raw output from ESLint.**

  72:5  error  Parsing error: Unexpected token )

✖ 1 problem (1 error, 0 warnings)

**Comment by platinumazure** [comment]
Hi @gamut, thanks for the issue.

The file looks incomplete. There are only about 50 lines in the file as well. Could you please provide the *complete* file contents being linted, as well as identify where line 72 is?

**Comment by gamut** [comment]
Done.  thx

**Comment by platinumazure** [comment]
Hi @gamut, thanks for posting the file.

There's a dangling comma after the last function argument on line 71. Dangling commas for function arguments is only valid in ES2017 or later. I'm not sure if the airbnb config might be using ES2017, but your config file definitely does not specify an ecmaVersion explicitly so it might be using ES5 by default. In any case, if you are willing to parse your code using ES2017 spec, you should add to your config:

```json
{
    "parserOptions": {
        "ecmaVersion": 2017
    }
}
```

**Comment by not-an-aardvark** [comment]
Closing because the question has been answered.

**Comment by gamut** [comment]
Thank you.  Appreciate the response.
